mesonbuild/dependencies/dev.py

```diff
@@ -700,6 +702,110 @@ def __init__(self, environment: 'Environment', kwargs: JNISystemDependencyKW):
 packages['jdk'] = JDKSystemDependency
 
 
+class DiaSDKSystemDependency(SystemDependency):
+
+    def _try_path(self, diadir: str, cpu: str) -> bool:
+        if not os.path.isdir(diadir):
+            return False
+
+        include = os.path.join(diadir, 'include')
+        if not os.path.isdir(include):
+            mlog.error('DIA SDK is missing include directory:', include)
+            return False
+
+        lib = os.path.join(diadir, 'lib', cpu, 'diaguids.lib')
+        if not os.path.exists(lib):
+            mlog.error('DIA SDK is missing library:', lib)
+            return False
+
+        bindir = os.path.join(diadir, 'bin', cpu)
+        if not os.path.exists(bindir):
+            mlog.error(f'Directory {bindir} not found')
+            return False
+
+        found = glob.glob(os.path.join(bindir, 'msdia*.dll'))
+        if not found:
+            mlog.error("Can't find msdia*.dll in " + bindir)
+            return False
+        if len(found) > 1:
+            mlog.error('Multiple msdia*.dll files found in ' + bindir)
+            return False
+        self.dll = found[0]
+
+        # Parse only major version from DLL name (eg '8' from 'msdia80.dll', '14' from 'msdia140.dll', etc.).
+        # Minor version is not reflected in the DLL name, instead '0' is always used.
+        # Aside from major version in DLL name, the SDK version is not visible to user anywhere.
+        # The only place where the full version is stored, seems to be the Version field in msdia*.dll resources.
+        dllname = os.path.basename(self.dll)
+        versionstr = dllname[len('msdia'):][:-len('.dll')]
+        if versionstr[-1] == '0':
+            self.version = versionstr[:-1]
+        else:
+            mlog.error(f"Unexpected DIA SDK version string in '{dllname}'")
+            self.version = 'unknown'
+
+        self.compile_args.append('-I' + include)
+        self.link_args.append(lib)
+        self.is_found = True
+        return True
+
+    # Check if compiler has a built-in macro defined
+    @staticmethod
+    def _has_define(compiler: 'Compiler', dname: str, env: 'Environment') -> bool:
+        defval, _ = compiler.get_define(dname, '', env, [], [])
+        return defval is not None
+
+    def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]) -> None:
```

Comment:
```suggestion
    def __init__(self, environment: Environment, kwargs: T.Dict[str, T.Any]) -> None:
```

Reply:
@dcbaker Is `T.Any` ok in this context?

Reply:
Full disclosure: This is the first time I am dealing with python type checking. I simply copied type descriptions from constructors of other *Dependency classes. Many existing classes have same declaration of kwargs.

Reply:
The quotes around Environment are because many other constructors were written before we were able to add `from __future__ import annotations` to the top of every file.

With python 3.7 as the minimum requirement, that future import is available and makes every type checking annotation act *as though* they are quoted:
- they don't have to be imported outside of `T.TYPE_CHECKING` to be used
- they don't create code objects at runtime, which means it's a bit faster to import modules

And therefore, you can also save two characters worth of space toward the max line length by skipping the quote characters too, as they are now redundant. :) But rewriting existing code causes a bit of churn so it's not obvious that it's worth doing so just to delete a couple redundant quotes.